client/test/lib.spec.js

```diff
@@ -89,6 +89,173 @@ describe('client', () => {
+        assert.unreachable('sholud have failed')
+      } catch (error) {
+        assert.ok(error instanceof TypeError)
+        assert.match(error, /proprety `image` must be a Blob or File/)
+      }
+    })
+
+    it('requires image mime type', async () => {
+      const client = new NFTStorage({ token, endpoint })
+      try {
+        await client.store({
+          name: 'name',
+          description: 'stuff',
+          image: new Blob(['bla bla']),
+        })
+      } catch (error) {
+        assert.ok(error instanceof TypeError)
+        assert.match(error, /Blob or File object with `image\/\*` mime type/)
+      }
+    })
+
+    it('expects decimal to be an int', async () => {
+      const client = new NFTStorage({ token, endpoint })
+      try {
+        await client.store({
+          name: 'name',
+          description: 'stuff',
+          image: new Blob(['pretend image'], { type: 'image/png' }),
+          // @ts-expect-error
+          decimals: 'foo',
+        })
+      } catch (error) {
+        assert.ok(error instanceof TypeError)
+        assert.match(error, /proprety `decimals` must be an integer value/)
+      }
+    })
+
+    it('errors without token', async () => {
+      const client = new NFTStorage({ token: 'wrong', endpoint })
+
+      try {
+        await client.store({
+          name: 'name',
+          description: 'tada',
+          image: new Blob([], { type: 'image/png' }),
+        })
+        assert.unreachable('sholud have failed')
+      } catch (error) {
+        assert.ok(error instanceof Error)
+        assert.match(error, /Unauthorized/)
+      }
+    })
+
+    it('uploads image', async () => {
+      const client = new NFTStorage({ token, endpoint })
+      const result = await client.store({
+        name: 'name',
+        description: 'stuff',
+        image: new Blob(['fake image'], { type: 'image/png' }),
+      })
+
+      assert.ok(result.metadata instanceof URL)
+      assert.ok(result.metadata.protocol, 'ipfs:')
+
+      assert.ok(result.ipld instanceof CID)
+
+      assert.equal(result.data.name, 'name')
+      assert.equal(result.data.description, 'stuff')
+      assert.ok(result.data.image instanceof URL)
+      assert.ok(result.data.image.protocol, 'ipfs:')
+
+      assert.equal(result.embed.name, 'name')
+      assert.equal(result.embed.description, 'stuff')
+      assert.ok(result.embed.image instanceof URL)
+      assert.ok(result.embed.image.protocol, 'https:')
+    })
+
+    it('store with properties', async () => {
+      const client = new NFTStorage({ token, endpoint })
+      const result = await client.store({
+        name: 'name',
+        description: 'stuff',
+        image: new File(['fake image'], 'cat.png', { type: 'image/png' }),
+        properties: {
+          extra: 'meta',
+          src: [
+            new File(['hello'], 'hello.txt', { type: 'text/plain' }),
+            new Blob(['bye']),
+          ],
+        },
+      })
```

Comment:
const data = new FormData()
data.set('image', File())
data.set('properties.src.0', new File())
data.set('properties.src.1': new Blob())
data.set('meta', JSON.stringify(metadata))
```

Reply:
Added this example to jsdoc comment of `Token.encode`